tests/pattern-matching-tests.lisp

```diff
@@ -73,3 +73,32 @@
   (is (match #\c
         (#\c True)
         (_ False))))
```

Comment:
We need to add a couple parser tests in the list, and show that good error messages happen